server/athenian/api/preloading/cache.py

```diff
@@ -113,25 +115,26 @@ def filter(
         :param uncast: whether to uncast the columns dtypes, see the `_squeeze` method.
 
         """
-        df = self._df.take(np.flatnonzero(mask))
-        if columns:
-            df = df[columns]
-        if uncast:
-            for col in self._identifier_cols:
-                try:
-                    df[col.key] = df[col.key].apply(lambda s: s.decode("utf8"))
-                except KeyError:
-                    continue
-
-            for col in self._categorical_cols:
-                try:
-                    df[col.key] = df[col.key].astype("object").replace({np.nan: None})
-                except KeyError:
-                    continue
-        if index:
-            df.set_index(index, inplace=True)
-
-        return df
+        with sentry_sdk.start_span(op=f"CachedDataFrame.filter/{self._id}"):
+            df = self._df.take(np.flatnonzero(mask))
+            if columns:
+                df = df[columns]
+            if uncast:
+                for col in self._identifier_cols:
+                    try:
+                        df[col.key] = df[col.key].apply(lambda s: s.decode("utf8"))
+                    except KeyError:
+                        continue
+
+                for col in self._categorical_cols:
+                    try:
+                        df[col.key] = df[col.key].astype("object").replace({np.nan: None})
```

Comment:
Unfortunately, the reason why I did it like this, it's because when `fillna` is called on a series that has dtype "category" the `method` param for `fillna` becomes required, I didn't investigate the reason though.

Reply:
```python
f[col.key].astype("object").fillna(None)
```
then